cmd/clairctl/clair/clair.go

```diff
@@ -0,0 +1,289 @@
+  result.Low = 0
+	result.Negligible = 0
+  
+  for _, l := range imageAnalysis.Layers {
+    for _, f := range l.Layer.Features {
+      result.Total += len(f.Vulnerabilities)
+      for _, v := range f.Vulnerabilities {
+        switch v.Severity {
+        case "High":
+          result.High++
+        case "Medium":
+          result.Medium++
+        case "Low":
+          result.Low++
+				case "Negligible":
+					result.Negligible++
+        }
+      }
+    }
+  }
+  
+  return result;
+}
+
+// Vulnerability : A vulnerability inteface
+type Vulnerability struct {
+	Name, Severity, IntroduceBy, Description, Link, Layer string
+}
+
+// Weight get the weight of the vulnerability according to its Severity
+func (v Vulnerability) Weight() int  {
+	weight := 0
+	
+	switch v.Severity {
+		case "High":
+			weight = 4
+		case "Medium":
+			weight = 3
+		case "Low":
+			weight = 2
+		case "Negligible":
+			weight = 1
+		}
+	
+	return weight
+}
+
+// Layer : A layer inteface
+type Layer struct {
+	Name string
+	Path string
+	Namespace string
+	Features []Feature
+}
+
+// Feature : A feature inteface
+type Feature struct {
+	Name string
+	Version string
+	Vulnerabilities []Vulnerability
+}
+
+// Status give the healthy / unhealthy statut of a feature
+func (feature Feature) Status() bool  {
+	return len(feature.Vulnerabilities) == 0;
+}
+
+// Weight git the weight of a featrure according to its vulnerabilities
+func (feature Feature) Weight() int {
+	weight := 0
+	
+	for _, v := range feature.Vulnerabilities {
+		weight += v.Weight()
+	}
+	
+	return weight
+}
+
+// VulnerabilitiesBySeverity sorting vulnerabilities by severity
+type VulnerabilitiesBySeverity []Vulnerability
+
+func (a VulnerabilitiesBySeverity) Len() int { return len(a) }
+func (a VulnerabilitiesBySeverity) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a VulnerabilitiesBySeverity) Less(i, j int) bool  {
+	return a[i].Weight() > a[j].Weight()
+}
+
+// LayerByVulnerabilities sorting of layers by global vulnerability
+type LayerByVulnerabilities []Layer
+
+func (a LayerByVulnerabilities) Len() int { return len(a) }
+func (a LayerByVulnerabilities) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a LayerByVulnerabilities) Less(i, j int) bool {
+	firstVulnerabilities := 0
+	secondVulnerabilities := 0
+	
+	for _, l := range a[i].Features {
+		firstVulnerabilities = firstVulnerabilities + l.Weight()
+	}
+	
+	for _ , l := range a[j].Features {
+		secondVulnerabilities = secondVulnerabilities + l.Weight()
+	}
+	
+	return firstVulnerabilities > secondVulnerabilities
+}
+
+// FeatureByVulnerabilities sorting off features by vulnerabilities
+type FeatureByVulnerabilities []Feature
+
+func (a FeatureByVulnerabilities) Len() int { return len(a) }
+func (a FeatureByVulnerabilities) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func (a FeatureByVulnerabilities) Less(i, j int) bool {
+	return a[i].Weight() > a[j].Weight()
+}
+
+// SortLayers give layers ordered by vulnerability algorithm
```

Comment:
Also, this doesn't make sense if the first layer adds 10 vulns and the second one removes them all. At the end, the image has no vulnerability. clairctl should only get the report for the latest layer. Then, it can use the `AddedBy` field in the `Features` to attribute the fact that this vulnerability has been introduced by a particular layer. But again, I think we should be really careful about how we display the layers.